### PR TITLE
Allow to set log level on Msg and Log

### PR DESCRIPTION
**Pull request by Kryptos-FR**
```clj
42 (Log "prefix") ; default LogLevel.Info
-1 (Log "ERROR" LogLevel.Error)

(Msg "Trace me" LogLevel.Trace)
(Msg "TODO" LogLevel.Debug)
(Msg "That's incorrect" :Level LogLevel.Warning)
```

In addition, logs are now displayed from `Maybe` else clause when silent:
```clj
; before
[0]
(Maybe
   (-> (Take 1))
   (-> (Msg "Cannot retrieve item at index 1"))
  :Silent true)
;; nothing displayed

; after
[0]
(Maybe
   (-> (Take 1))
   (-> (Msg "Cannot retrieve item at index 1" :Level LogLevel.Error)
  :Silent true)
;; [error] [2022-11-18 14:48:26.889] [T-33400] [logging.cpp::101] [main-wire] Cannot retrieve item at index 1
```

### CODE REVIEW

File: src/core/shards/flow.cpp [hunk 1]
```diff
@@ -456,20 +456,18 @@ struct Maybe : public BaseSubFlow {
       if (_silent) {
         spdlog::set_level(spdlog::level::off);
       }
-      DEFER({
-        if (_silent) {
+      auto state = _shards.activate(context, input, output);
+      if (_silent) {
 #ifdef NDEBUG
```
Comment (guusw) [inline src/core/shards/flow.cpp:461]: Might be a good opportunity to simplilfy this, just store `spdlog::get_level()` and restore it after activate

File: src/mal/SHCore.cpp [hunk 2]
```diff
@@ -1800,14 +1800,30 @@ BUILTIN("context-var") {
...
+template <> struct GetComponentType<SHType::Float2> {
+  typedef double Type;
+};
+template <> struct GetComponentType<SHType::Float3> {
+  typedef float Type;
+};
+template <> struct GetComponentType<SHType::Float4> {
+  typedef float Type;
+};
+template <> struct GetComponentType<SHType::Int2> {
```
Comment (guusw) [inline src/mal/SHCore.cpp:1812]: Did you change formatting settings, if so can you commit those too? 
for me it puts these on one line using the current .clang-format rules.

**Review by guusw — CHANGES_REQUESTED** [review]

File: src/mal/SHCore.cpp (hunk 2 above)
Comment (Kryptos-FR) [reply src/mal/SHCore.cpp:1812]: I didn't change the formatting settings. I ran `cargo fmt` and `ninja format`.

**Review by sinkingsugar — APPROVED** [review]
Nice 👍

**Pull request merged**